hearpreprocess/pipeline.py

```diff
@@ -474,12 +486,51 @@ def run(self):
         metadata = self.get_all_metadata()
         print(f"metadata length = {len(metadata)}")
 
+        def _log_metadata_split_size_label(metadata, event_str):
+            diagnostics.info(
+                f"{self.longname} Files in each split {event_str}: "
+                "{}".format(metadata.groupby("split")["relpath"].nunique().to_dict())
+            )
+            diagnostics.info(
+                f"{self.longname} Unique labels in each split {event_str}: "
+                "{}".format(metadata.groupby("split")["label"].nunique().to_dict())
+            )
+
+        _log_metadata_split_size_label(metadata, "Before Postprocessing")
         metadata = self.postprocess_all_metadata(metadata)
+        _log_metadata_split_size_label(metadata, "After Postprocessing")
 
         # Split the metadata to create valid and test set from train if they are not
         # created explicitly in get_all_metadata
         metadata = self.split_train_test_val(metadata)
 
+        # Each split should have unique files and no file should be across splits
+        assert (
+            len(
+                set.intersection(
+                    *map(set, metadata.groupby("split")["relpath"].apply(list).tolist())
+                )
+            )
+            == 0
+        ), "Relpath is not unique across split"
```

Comment:
Why not do this for unique_filestem, which is more important than relpath?

Reply:
I mean yes. we can do that. but we have asserted before a one-to-one correspondance between uniquefilestem and relpath. I can change it though.

Reply:
Then assert the one that is more important, which is filestem